src/zarr/abc/store.py

```diff
@@ -329,16 +329,22 @@ def supports_listing(self) -> bool:
         ...
 
     @abstractmethod
-    def list(self) -> AsyncGenerator[str]:
+    def list(self) -> AsyncIterator[str]:
         """Retrieve all keys in the store.
 
         Returns
         -------
-        AsyncGenerator[str, None]
+        AsyncIterator[str]
+
+        Notes
+        -----
+        This method should be async,
+        `How to correctly specify type hints with AsyncGenerator and AsyncContextManager <https://stackoverflow.com/questions/68905848/how-to-correctly-specify-type-hints-with-asyncgenerator-and-asynccontextmanager>`_
+        explains why it is not.
```

Comment:
I'd like to move this to a comment rather than the public docstring. Most implementations will inherit this and end users don't need to know about this implementation detail.

Reply:
Yes, I wasn't sure about it. I put it in the documentation because it documents an inconsistency in function signatures.

Reply:
Ideally, I would document this using something similar to disabling Pylint warning [W0236](https://pylint.readthedocs.io/en/latest/user_guide/messages/warning/invalid-overridden-method.html):
```
# pylint: disable=invalid-overridden-method
```
Unfortunately there's no equivalent rule in ruff, as far as I can see.

Reply:
Fixed.

Reply:
Thanks.